e2e/testcases/helm_sync_test.go

```diff
@@ -380,3 +409,75 @@ func helmChartVersion(chartVersion string) nomostest.Sha1Func {
 		return chartVersion, nil
 	}
 }
+
+// pushes a new helm chart for use during an e2e test
+func pushHelmChart(nt *nomostest.NT, t *testing.T, helmchart, version string) string {
+	t.Helper()
+
+	nt.T.Log("Push helm chart to the artifact registry")
+	tmpDir := t.TempDir()
+
+	chartName := fmt.Sprintf("%s-%s-%s-%s", helmchart, nomostesting.GCPProjectIDFromEnv, nomostesting.GCPClusterFromEnv, generateRandomString())
+	if len(chartName) > 50 {
+		// the chartName + releaseName is used as the metadata.name of resources in the coredns helm chart, so we must trim this down
+		// to keep it under the k8s length limit
+		chartName = chartName[len(chartName)-50:]
+		chartName = strings.Trim(chartName, "-")
+	}
+
+	paths, err := filepath.Glob(fmt.Sprintf("../testdata/helm-charts/%s/*", helmchart))
+	if err != nil {
+		nt.T.Fatalf("failed to get chart filepaths: %v", err)
+	}
+	for _, path := range paths {
+		if _, err := nt.Shell.Command("cp", "-r", path, tmpDir).CombinedOutput(); err != nil {
+			nt.T.Fatalf("failed to copy helm chart: %v", err)
+		}
+	}
+	if _, err := nt.Shell.Command("sed", "-i", "-e", fmt.Sprintf("s/name: %s/name: %s/g", helmchart, chartName), filepath.Join(tmpDir, "Chart.yaml")).CombinedOutput(); err != nil {
+		nt.T.Fatalf("failed to rename helm chart: %v", err)
+	}
+	if _, err := nt.Shell.Helm("package", tmpDir, "--destination", tmpDir); err != nil {
```

Comment:
For each of these `helm` subcommands I think it would be nice to have a layer of abstraction (i.e. a function/method). I think that would help with reuse/readability

Reply:
Moved it out to another file with an abstraction layer!